### ISSUE TITLE
AES-CTR "testType": "CTR" payload length

### ISSUE THREAD

**Issue by rnunez83** [usnistgov/ACVP#1234]
**Protocol Section**
I have a questions regarding the AES-CTR testype "CTR". 

In [1] it states that for "CTR" - Counter Mode Test ->  "The server will send a long message to the client for encryption or decryption and back-compute the IVs used by the implementation under test."

Furthermore, in [2] it states that for AES-CTR the payload length "Min": 1, "Max": 128, "Inc": any

Finally, in [3] the payloadlen in the example shown indicates a payload of 12800 while in my own testing seems to always generates 12800**0** bits

--snip from [3]
{
    "tgId": 3,
    "direction": "encrypt",
    "keyLen": 128,
    "testType": "CTR",
    "tests": [{
      "tcId": 829,
      "key": "3A9A8485E1B7BA1987F88F8C095257C4",
      **"payloadLen": 12800,**
      "pt": "CE8E4B6F7C68DE5FDE3..."
    }
--end snip

Examples:
[1] https://github.com/usnistgov/ACVP/blob/master/src/symmetric/sections/04-testtypes.adoc
[2] https://github.com/usnistgov/ACVP/blob/master/src/symmetric/sections/05-capabilities.adoc#property_grid_misc
[3] https://github.com/usnistgov/ACVP/blob/master/src/symmetric/sections/97-examples.adoc

**Protocol Question**
My questions is what is the length of the "payload" that will be used for the "testtype":"CTR"? 

Is this value fixed or is it possible to configure to set to the max an IUT can support?

**Comment by rnunez83** [comment]
Adding further findings:

I tried to increase the payload size larger than 128-bits and received the following error message from the server:

ACVPProxy (11:36:59) (tid0) Error: Server error response: [
  {
    "acvVersion": "1.0"
  },
  {
    "error": "Validation error(s) on JSON payload.",
    "context": [
      "ACVP-AES-CTR-1.0: DataLength maximum must be at most 128"
    ]
  }
]

**Comment by livebe01** [comment]
Yes, as I've been looking at this it appears that we require an implementation to support a 128-bit payload length as AES at its core operates on 128-bit blocks. If an implementation supports payloadLens < 128-bits, we can test that as well. 

So, when you do a registration, if you want to test payloadLens other than 128-bits, you'll see one or more test groups appear in your prompt such that the test cases contain the specific *payloadLen* size that's being tested. For all other test cases in the prompt, test cases that do not specify their *payloadLen*, the *payloadLen* is the 128-bit core block size that AES uses. 

What you're seeing with the large/128,000-bit pt's or ct's for the "CTR" tests is a large message that needs to be encrypted or decrypted. This 128,000-bit message will need to be broken into the default 128-bit chunks and each 128-bit chunk of the message encrypted or decrypted.

Our apologies for the bogus *payloadLen*s that showed up in the AES-CTR examples. I'll clean those up so they don't cause any further confusion

**Comment by rnunez83** [comment]
Thank you for looking into this and thank you clarifying.

So if I understand correctly, for TestType: CTR, CT and PT values will always be 128,000-bits and the IUT is expected to process in 128-bit chunks to form a 128k-bit output in the response file.

**Comment by livebe01** [comment]
Yes, that should be the case. However, test harnesses and implementations should be designed in a manner such that they will be able to accommodate a different sized payload (multiple of 128 bits) if we were to change the CTR payload size for any reason in the future.
